src/interfaces/ICirculator.sol

```diff
@@ -43,13 +64,15 @@ interface ICirculator {
     /// @param s Signature s.
     struct DelegateData {
         uint32 destinationDomain;
-        bytes32 recipient;
+        uint32 fillDeadline;
+        address recipient;
+        uint256 outputAmount;
```

Comment:
it makes more sense to add the `mode` here, and also include it in the signature.

(all inputs should be signed by the user)